classes.py

```diff
@@ -0,0 +1,82 @@
+import datetime
+from dataclasses import dataclass
+# from mpydev import BioPac as bp
+import numpy as np
+from datetime import date
+
+
+@dataclass
+class Subject:
+    Subject_ID: int
+    Subject_ID = '0'
+    Subject_Test_Date = date.today()
+    Subject_Key = f'{Subject_ID}({Subject_Test_Date})'
+
+
+class SubjectHandle(Subject):
+    def identify_subject(self):
+        subject_id = input('Insert subject ID')
+        if isinstance(subject_id, int):
+            self.Subject_ID = subject_id
+        else:
+            raise ValueError('The Subjects ID must be an integer')
+            self.identify_subject()
+
+    def get_subject_id(self):
+        return self.Subject_ID
+
+    def get_subject(self):
+        return self.Subject_Key
+
+
+@dataclass
+class TestData(SubjectHandle):
+    Test_Data: dict
+
+
+class TestDataHandler(TestData):
+
+    def initialize_test(self):
+        self.identify_subject()
+        key = self.Subject_Key()
+        self.Test_Data = {key: {}}
+
+    def first_timestamp(self, first_timestamp_list: dict):
+        self.Test_Data[self.get_subject()]['Timestamps'] = first_timestamp_list
+
+    def timestamp_add(self, timestamp: dict):
+        self.Test_Data[self.get_subject()]['Timestamps'].update(timestamp)
+
+    def database_commit(self):
+        with open(f'{self.get_subject()}.txt') as json:
+            json.dump(self.Test_Data)
+
+
+@dataclass
+class TimeStamps(TestDataHandler):
+    timestamp: int or float or str
+    ts_dict: dict
+    stage: str
+    one_to_twenty_four: list
+    one_to_twenty: list
+    one_to_twenty_four = np.arange(1, 25, 1)
+    twenty_seven_to_eighty_one = np.arange(27, 81, 1)
+    allowed_stages = ['Baseline', f'Q{one_to_twenty_four}', f'IAPS{twenty_seven_to_eighty_one}']
+
+
+class ManageTimestamps(TimeStamps):
+    def create_timestamp(self, stage: str):
+        if stage in self.allowed_stages:
```

Comment:
I'd always encourage fail first where possible, it removes a level of indention and removes the requirement of an else statement

```python
if stage not in self.allowed_stages:
    raise ValueError('Stage must be one of the pre-approved for this protocol.')

self.stage = stage
```

Reply:
But we're removing that exception in favour of a boolean, right?

Reply:
@pedrorribeiro That was for the identify subject method, an exception here makes more sense because if it's trying to add a timestamp at a stage which isn't allowed, we want the application to fail (or the exception to be caught, and the application to correct itself).

Going by your description of baseline, I'll have another look if this is needed.